packages/bierzo-wallet/src/routes/welcome/index.e2e.spec.ts

```diff
@@ -0,0 +1,54 @@
+/*global chrome*/
+import { Browser, Page } from 'puppeteer';
+//import { WELCOME_ROUTE } from '../paths';
+import {
+  launchBrowser,
+  createPage,
+  createExtensionPage,
+  closeBrowser,
+  getBackgroundPage,
+} from '../../utils/test/e2e';
+import { submitExtensionSignupForm } from './test/fillSignupForm';
+import { withChainsDescribe } from '../../utils/test/testExecutor';
+
+withChainsDescribe(
+  'DOM > Welcome route',
+  (): void => {
+    let browser: Browser;
+    let page: Page;
+    let backgroundPage: Page;
+    let extensionPage: Page;
+
+    beforeEach(async (): Promise<void> => {
+      browser = await launchBrowser();
+      page = await createPage(browser);
+      extensionPage = await createExtensionPage(browser);
+      backgroundPage = await getBackgroundPage(browser);
```

Comment:
I like the `backgroundPage` approach. The initial approach I had in mind was:
```
const badgeText = await page.evaluate(    <------ Note page here
    (): Promise<string | undefined> => {
        return new Promise(resolve => {
            chrome.browserAction.getBadgeText({}, text => resolve(text));
        });
   },
);
```

I thought having the page tab with the extension installed you already had access to `chrome.browserAction`.

But the `getBackgroundPage` is so cool. It is so damn cool that despite we can have other alternatives we are going to leave it in the repo. 

Just curiosity, did you try my approach?

Reply:
Yes I tried it, but you do not have access to `chome.browserAction` from it.